apps/vscode-extension/src/intellisense/design-token.ts

```diff
@@ -0,0 +1,72 @@
+import { CompletionItem, CompletionItemKind, CompletionItemProvider, HoverProvider } from 'vscode';
+import { DesignTokenVariableSet, DesignTokenVariableStructure, getCssTokenValueRenderer, parseDesignTokenFile } from '@o3r/design';
+import * as vscode from 'vscode';
+import * as path from 'node:path';
+
+const readFile = async (filePat: string) => (await vscode.workspace.fs.readFile(vscode.Uri.parse(filePat))).toString();
+
+const getDesignTokens = async (currentFile?: string) => {
```

Comment:
I think you should add a cache mechanism for the token calculation for the case we will have a large amount of files.
A simple way it to recalculate the Design Token only if one of the file of the glob has changed

Reply:
Will be done in another [task](https://github.com/AmadeusITGroup/otter/issues/1490)